service/controller/resource/clustersecret/desired.go

```diff
@@ -0,0 +1,80 @@
+package clustersecret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/k8smetadata/pkg/annotation"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/cluster-apps-operator/pkg/project"
+	"github.com/giantswarm/cluster-apps-operator/service/controller/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*corev1.Secret, error) {
+	if !r.enabled {
+		return nil, nil
+	}
+
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var secrets []*corev1.Secret
+
+	if key.IsDeleted(&cr) {
+		r.logger.Debugf(ctx, "deleting cluster secrets for cluster '%s/%s'", cr.GetNamespace(), key.ClusterID(&cr))
+		return secrets, nil
+	}
+
+	secretSpecs := []secretSpec{
+		{
+			Name:      key.ClusterValuesResourceName(&cr),
+			Namespace: key.ClusterID(&cr),
+			Values:    map[string]interface{}{},
+		},
+	}
```

Comment:
If this secret is only needed for CAPO then I would add a provider flag to the resource. So we only create it there. WDYT?

Reply:
Good point. I made this change.